core/src/core/transaction.rs

```diff
@@ -431,14 +431,16 @@ impl Transaction {
 	/// Validates all relevant parts of a fully built transaction. Checks the
 	/// excess value against the signature as well as range proofs for each
 	/// output.
-	pub fn validate(&self) -> Result<(), Error> {
+	pub fn validate(&self, as_block: bool) -> Result<(), Error> {
 		if self.inputs.len() > consensus::MAX_BLOCK_INPUTS {
 			return Err(Error::TooManyInputs);
 		}
-		self.verify_features()?;
+		if !as_block {
```

Comment:
Not sure how critical this is but I've been leaning toward ordering these verification rules _roughly_ based on how expensive they might be to run.

Reply:
I think that's a good idea. I was trying to avoid checking that bool multiple times, maybe something we can fix up as well when we remove it?